hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -632,6 +658,19 @@ private void processDataBlock(HoodieDataBlock dataBlock, Option<KeySpec> keySpec
       while (recordIterator.hasNext()) {
         IndexedRecord currentRecord = recordIterator.next();
         IndexedRecord record = schemaOption.isPresent() ? HoodieAvroUtils.rewriteRecordWithNewSchema(currentRecord, schemaOption.get(), Collections.emptyMap()) : currentRecord;
+
+        Schema schema = record.getSchema();
+        if (this.dropPartitions && this.partitionFields.isPresent() && this.partitionValues.isPresent()) {
+          String[] partitionFields = this.partitionFields.get();
+          String[] partitionValues = this.partitionValues.get();
+
+          if (partitionFields.length == partitionValues.length) {
+            for (int i = 0; i < partitionValues.length; i++) {
+              record.put(schema.getField(partitionFields[i]).pos(), partitionValues[i]);
```

Comment:
We can't just blindly inject strings into record, we need to make sure we coercing it to a type of the column